docs/src/components/file-tree.astro

```diff
@@ -57,7 +57,7 @@ const processedContent = await fileTreeProcessor.process({
 		color: var(--sl-color-gray-3);
 	}
 
-	file-tree .directory > details > summary:hover .tree-icon,
+	file-tree .directory > details > summary:hover > :not(.highlight) > .tree-icon,
```

Comment:
Looks like this is always disabling the hover styling for the tree icon because the DOM structure is not `summary > .highlight > .tree-icon` but more nested:

<img width="318" alt="DOM structure showing additional nesting between summary, highlight and tree-icon" src="https://github.com/withastro/starlight/assets/357379/765b3d93-d50b-4945-9853-47775682e465">

Maybe it’s simpler to just add an override style something like this?

```css
file-tree .directory > details > summary:hover .highlight .tree-icon {
	fill: var(--sl-color-text-invert);
}
```

Reply:
Good catch, I was focused on the hovered style and I missed the regression. I've changed the fix to use your version which was the alternative method I was thinking about initially. Tried to avoid it with a `:not` approach to gather together the rules, but it's prolly clearer to do it like this :)